### PR TITLE
test: port twonode test to Go, remove multinode test

### PR DESCRIPTION
**Pull request by guseggert**
The multinode test is effectively the same as the twonode test. There are some problems with it too: it *looks* like it's testing the Websocket transport with the "listentype,ws" IPTB attribute, but that attribute doesn't actually exist in ipfs/iptb-plugins, so it does nothing, so that test actually just runs the same test twice (Yamux disabled). Furthermore, this is just the same test as in the mplex twonode test. So this just removes the useless multinode test entirely. The multinode test takes around 130 seconds on average, so this is a big win in terms of execution time.

Also, this removes the part of the twonode test that checks the amount of data transferred over Bitswap. This is an implementation detail of Bitswap, it's not appropriate to test this in an end-to-end test as it depends on algorithmic details of how Bitswap works, and has nothing to do with transports. This is probably more appropriate as a perf or benchmark test of Bitswap. This is further illustrated by the fact that this check is flaky.

This also moves equivalent functionality from jbenet/go-random-files into the testutils package. This just copies the code and modifies it slightly for better ergonomics.

### CODE REVIEW

File: test/cli/testutils/random_files.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path"
+	"time"
+)
+
+var AlphabetEasy = []rune("abcdefghijklmnopqrstuvwxyz01234567890-_")
+var AlphabetHard = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890!@#$%^&*()-_+= ;.,<>'\"[]{}() ")
```
Comment (Jorropo) [inline test/cli/testutils/random_files.go:13]: Kubo is meant to be blind(ish) to the content of files, it is bytes in bytes out regardless of content.
I expect testing with `/dev/urandom` data. Else we could hide bugs either in kubo or in harness.

File: test/cli/transports_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+				cfg.Routing.Type = config.NewOptionalString("none")
+				cfg.Bootstrap = nil
+			})
+		})
+	}
+	checkSingleFile := func(nodes harness.Nodes) {
+		s := testutils.RandomStr(100)
+		hash := nodes[0].IPFSAddStr(s)
+		nodes.ForEachPar(func(n *harness.Node) {
+			val := n.IPFS("cat", hash).Stdout.Trimmed()
```
Comment (Jorropo) [inline test/cli/transports_test.go:28]: The `.Trimmed` is (can be) hidding bugs here.

File: test/cli/transports_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	}
+	checkRandomDir := func(nodes harness.Nodes) {
+		randDir := filepath.Join(nodes[0].Dir, "foobar")
+		require.NoError(t, os.Mkdir(randDir, 0777))
+		rf := testutils.NewRandFiles()
+		rf.FanoutDirs = 3
+		rf.FanoutFiles = 6
+		require.NoError(t, rf.WriteRandomFiles(randDir, 4))
+
+		hash := nodes[1].IPFS("add", "-r", "-Q", randDir).Stdout.Trimmed()
```
Comment (Jorropo) [inline test/cli/transports_test.go:40]: `.Trimmed` too here.

**Review by Jorropo — CHANGES_REQUESTED** [review]
Mostly LGTM except `.Trimmed` issue.

File: test/cli/testutils/random_files.go (hunk 1 above)
Comment (guseggert) [reply test/cli/testutils/random_files.go:13]: These are not used for file contents, but for filenames, which have restricted character sets and reserved characters depending on the OS.  The contents of the files are indeed just random bytes. Also this is just a blind port from https://github.com/jbenet/go-random-files

File: test/cli/transports_test.go (hunk 2 above)
Comment (guseggert) [reply test/cli/transports_test.go:28]: thanks agreed that we shouldn't use Trimmed() on `ipfs cat`.

File: test/cli/transports_test.go (hunk 3 above)
Comment (guseggert) [reply test/cli/transports_test.go:40]: I think `Trimmed` is correct to use here, as the CLI writes a newline char to stdout that needs to be removed. This is not the case for `ipfs cat`. The sharness tests do the same thing, except the trimming happens in bash, which strips the trailing newline as part of command substitution.

**Review by Jorropo — APPROVED** [review]

**Comment by guseggert** [comment]
One of the Docker sharness tests is failing but it is unrelated; something changed on the GH Actions side, I have tested locally and it works fine. Will work on a fix separately.

**Pull request merged**